apax/layers/ntk_linear.py

```diff
@@ -5,31 +5,46 @@
 
 
 class NTKLinear(nn.Module):
+    """Linear layer with activation.
+    Corresponds to an NTK layer with "normal" and "zeros" for w and b initialization
+    and "use_ntk" set to True.
+    """
+
     units: int
-    b_init: str = "normal"
+    w_init: str = "normal"
+    b_init: str = "zeros"
+    use_ntk: bool = True
     dtype: Any = jnp.float32
 
     @nn.compact
     def __call__(self, inputs):
         inputs = inputs.astype(self.dtype)
 
-        w_initializer = nn.initializers.normal(1.0, dtype=self.dtype)
+        if self.w_init == "normal":
+            w_initializer = nn.initializers.normal(1.0, dtype=self.dtype)
+        elif self.w_init == "lecun":
+            w_initializer = nn.initializers.lecun_normal(dtype=self.dtype)
+        else:
+            raise NotImplementedError(f"Unknown weight initializer: {self.w_init}.")
```

Comment:
I think this should be a ValueError
see https://docs.python.org/3/library/exceptions.html#TypeError and https://docs.python.org/3/library/exceptions.html#NotImplementedError